LiteDB/Engine/Disk/MemoryCache.cs

```diff
@@ -401,28 +435,186 @@ private void Extend()
         public int Clear()
         {
             var counter = 0;
+            var now = DateTime.UtcNow;
+            var overCap = false;
 
             ENSURE(this.PagesInUse == 0, "must have no pages in use when call Clear() cache");
 
-            foreach (var page in _readable.Values)
+            // Thread-safe enumeration of ConcurrentDictionary
+            foreach (var kv in _readable)
             {
-                page.Position = long.MaxValue;
-                page.Origin = FileOrigin.None;
+                if (_readable.TryRemove(kv.Key, out var page))
+                {
+                    // Reset page controls
+                    page.Position = long.MaxValue;
+                    page.Origin = FileOrigin.None;
+
+                    // Mark as "freed at" for idle policy
+                    page.Timestamp = now.Ticks;
 
-                _free.Enqueue(page);
+                    _free.Enqueue(page);
+                    counter++;
 
-                counter++;
+                    // Check if we exceed free pages limit
+                    if (_free.Count > _maxFreePages)
+                        overCap = true;
+                }
             }
 
-            _readable.Clear();
+            // Trigger cleanup if needed (outside any locks)
+            if (overCap)
+            {
+                _cleanupSignal = true;
+                TryCleanupOnDemand(force: true);
+            }
 
             return counter;
         }
 
         #endregion
 
+        #region Memory Cleanup & Tuning
+
+        /// <summary>
+        /// Apply predefined cache tuning profiles for different environments
+        /// </summary>
+        public void ApplyProfile(CacheProfile profile)
+        {
+            switch (profile)
+            {
+                case CacheProfile.Mobile:
```

Comment:
I'd like some more "Dynamic" way: 
- CacheProfile becomes a struct or class
- CacheProfile has 3 static factory properties or readonly static fields if the nonstatic fields are readonly.
- CacheProfile can be configured with any of the 4 parameters mentioned (with checks)